plugins/AuditLog/AuditLog.php

```diff
@@ -238,19 +238,24 @@ public function beforeParticipantDelete()
                 return;
             }
 
-            $oNewParticipant=$this->getEvent()->get('model');
-            $oCurrentUser=$this->api->getCurrentUser();
+            $sTokenIds=$this->getEvent()->get('sTokenIds');
+            $aTokenIds = explode(',', $sTokenIds);
 
-            $aValues=$oNewParticipant->getAttributes();
+            foreach ($aTokenIds as $tokenId){
+                $token = Token::model($iSurveyID)->find('tid=' . $tokenId);
+                $oCurrentUser=$this->api->getCurrentUser();
```

Comment:
Updated.

2016-05-05 13:20 GMT+02:00 Olle Härstedt <notifications@github.com>:

> In plugins/AuditLog/AuditLog.php
> <https://github.com/LimeSurvey/LimeSurvey/pull/487#discussion_r62175178>:
>
> >
> > -            $aValues=$oNewParticipant->getAttributes();
> > +            foreach ($aTokenIds as $tokenId){
> > +                $token = Token::model($iSurveyID)->find('tid=' . $tokenId);
> > +                $oCurrentUser=$this->api->getCurrentUser();
>
> TIP: This can be moved outside the loop.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/LimeSurvey/LimeSurvey/pull/487/files/62311749e5a43fd99aeffdeba3ef5a5c1c05c7e2#r62175178>
>